packages/near-lake-primitives/README.md

```diff
@@ -0,0 +1,345 @@
+# NEAR Lake Primitive Types
+
+This crate contains the primitive types used by the [NEAR Lake Framework package](https://www.npmjs.com/package/@near-lake/framework). These types are used to define the data structures used by the framework as well as provide some popular helper functions.
+
+## Types
+
+### `Block`
+
+#### `Block` Structure Definition
+
+The `Block` type is used to represent a block in the NEAR Lake Framework. It is comprised by the following structure:
+
+```ts
+export class Block {
+    constructor(
+        readonly streamerMessage: StreamerMessage,
+        private executedReceipts: Receipt[],
+        readonly postponedReceipts: Receipt[],
+        readonly transactions: Transaction[],
+        private _actions: Map<string, Action>,
+        private _events: Map<string, Event[]>,
+        private _stateChanges: StateChange[]) { 
+
+        }
+    ... // helper methods and getters ommitted for brevity
+}
+```
+
+##### `streamerMessage`
+
+Low-level structure for backward compatibility. As implemented in previous versions of [`near-lake-framework`](https://www.npmjs.com/package/near-lake-framework). 
+
+##### `postponedReceipts`
+
+Receipts included on the chain but not executed yet marked as “postponed”: they are represented by the same structure `Receipt` (see the corresponding section in this doc for more details).
+
+##### `transactions`
+
+List of included `Transactions`, converted into `Receipts`.
+
+**_NOTE_:** Heads up! You might want to know about `Transactions` to know where the action chain has begun. Unlike Ethereum, where a Transaction contains everything you may want to know about a particular interaction on  the Ethereum blockchain, Near Protocol because of its asynchronous nature converts a `Transaction` into a `receipt` before executing it. Thus, On NEAR, `Receipts` are more important for figuring out what happened on-chain as a result of a Transaction signed by a user. Read more about [Transactions on Near](https://nomicon.io/RuntimeSpec/Transactions) here.
```

Comment:
```suggestion
**_NOTE_:** Heads up! You might want to know about `Transactions` to know where the action chain has begun. Unlike Ethereum, where a Transaction contains everything you may want to know about a particular interaction on  the Ethereum blockchain, Near Protocol because of its asynchronous nature converts a `Transaction` into a `Receipt` before executing it. Thus, On NEAR, `Receipts` are more important for figuring out what happened on-chain as a result of a Transaction signed by a user. Read more about [Transactions on Near](https://nomicon.io/RuntimeSpec/Transactions) here.
```